.github/workflows/tests.yml

```diff
@@ -1,5 +1,10 @@
 name: Run Tests
-on: [push]
+on:
+  push:
+    branches:
```

Comment:
WIth the previous `on push` we were obviously running each time a commit was pushed, do we have any concern about this? It feels like the right decision given it will run on PR or push to canary/main and prevent using more minutes on git actions.

Reply:
I initially had this as `on: [push, pull_request]` but when pushing changes to the PR, it triggered duplicate runs in the CI. So our configuration for this needs to make sure only one event triggers the CI to run at a time. This was the most common configuration I saw others using.